src/Microsoft.DotNet.Interactive.Http/HttpKernel.cs

```
@@ -318,6 +321,209 @@ private List<Diagnostic> GetAllDiagnostics(HttpRequestParseResult parseResult)
+        {
+            (int? Min, int? Max)? parameters = ParseParametersFromMatch(match);
+
+            if (parameters is not null)
+            {
+                int? min = parameters.Value.Min;
+                int? max = parameters.Value.Max;
+
+                if (!min.HasValue && !max.HasValue)
+                {
+                    text = string.Concat(text.Substring(0, match.Index), random.Next().ToString(), text.Substring(match.Index + match.Value.Length));
+                }
+                else if (!min.HasValue && max.HasValue)
+                {
+                    text = string.Concat(text.Substring(0, match.Index), random.Next(max.Value).ToString(), text.Substring(match.Index + match.Value.Length));
+                }
+                else if (min.HasValue && max.HasValue)
+                {
+                    text = string.Concat(text.Substring(0, match.Index), random.Next(min.Value, max.Value).ToString(), text.Substring(match.Index + match.Value.Length));
+                }
+            }
+
+            return node.CreateBindingSuccess(text);
+        } else
+        {
+            return node.CreateBindingFailure(HttpDiagnostics.CannotResolveSymbol(text));
+        }
+        
+    }
+
+    internal static (int? min, int? max)? ParseParametersFromMatch(Match match)
+    {
+        if (match.Success)
+        {
+            Group group = match.Groups["parameters"];
+            if (group.Captures.Count == 0)
+            {
+                return (null, null);
+            }
+            else if (group.Captures.Count == 1
+                && int.TryParse(group.Captures[0].Value, out int max))
+            {
+                return (null, max);
+            }
+            else if (group.Captures.Count == 2
+                && int.TryParse(group.Captures[0].Value, out int min)
+                && int.TryParse(group.Captures[1].Value, out max)
+                && min <= max)
+            {
+                return (min, max);
+            }
```

Should we report an error if there are more than 2 matches?

Also consider using a switch statement instead of the if else.